realm/realm-library/src/main/java/io/realm/internal/SharedGroup.java

```diff
@@ -293,6 +297,29 @@ public int hashCode() {
         }
     }
 
+    /**
+     * Waits for change committed by {@link SharedGroup} in other Thread.
+     *
+     * @return {@code true} if successfully detects change, {@code false} no change has detected otherwise.
+     */
+    public boolean waitForChange() {
+        if (isStoppedWaitingByUser) {
+            isStoppedWaitingByUser = false;
+            return false;
+        }
+        boolean changeStatus = nativeWaitForChange(nativePtr);
```

Comment:
Much of this would be cleaned. Lots of recent commits are about demonstrating possibilities to come up with a clean-slated solution. Since we've reached one, this will be cleaned.